### PR TITLE
added jmx username and password support

### PR DESCRIPTION
**Pull request by balajivenki**
Added support for jmx username and password.

### CODE REVIEW

**Comment by AntonGabov** [comment]
Hello, Balaji!
Thank you for your contribution to ctop! We appreciate it.

Don't forget to update README file with new info concerning jmx credentials.

File: src/main/java/com/pixonic/ctop/Main.java [hunk 1]
```diff
@@ -8,26 +8,42 @@
...
         }
 
         String hostAndPort = args[0];
         String keySpace = args[1];
+
         int interval = 10;
         if (args.length > 2) {
             interval = Integer.parseInt(args[2]);
         }
+
```
Comment (AntonGabov) [inline src/main/java/com/pixonic/ctop/Main.java:28]: If we pass arguments as “host:20000 key_space user:password”, NumberFormatException will be thrown.

File: src/main/java/com/pixonic/ctop/Main.java [hunk 2]
```diff
@@ -8,26 +8,42 @@
...
             interval = Integer.parseInt(args[2]);
         }
+        
+        boolean hasCreds = false;
+        String[] creds = null;
+        if (args.length > 3) {
+            creds = args[3].trim().split(":", 2);
+            hasCreds = true;
+        }
+
```
Comment (AntonGabov) [inline src/main/java/com/pixonic/ctop/Main.java:35]: Redundant empty lines.

File: src/main/java/com/pixonic/ctop/Main.java [hunk 3]
```diff
@@ -8,26 +8,42 @@
...
 
         System.out.println("Connecting to " + hostAndPort + "...");
 
+
```
Comment (AntonGabov) [inline src/main/java/com/pixonic/ctop/Main.java:39]: Redundant empty line.

File: src/main/java/com/pixonic/ctop/Main.java [hunk 4]
```diff
@@ -8,26 +8,42 @@
...
         final JMXServiceURL target = new JMXServiceURL("service:jmx:rmi:///jndi/rmi://" + hostAndPort + "/jmxrmi");
-        final JMXConnector connector = JMXConnectorFactory.connect(target);
+
+
+        HashMap environment = new HashMap();
```
Comment (AntonGabov) [inline src/main/java/com/pixonic/ctop/Main.java:43]: I would better use 
`Map<String, Object> environment  = new HashMap<>();`

**Comment by balajivenki** [comment]
@AntonGabov Thanks for your comments. I am actually working on making all these loaded from a properties file along with a mechanism to push the data to Graphite on a flag basis. I will fix your code review comments and send a final PR once my changes are done. Please let me know if the plan sounds good to you.

**Comment by balajivenki** [comment]
Closing this as I will send a new PR with all my changes

**Comment by AntonGabov** [comment]
Ok, sure. Your plan sounds good!

**Pull request closed without merging**